pkg/aws/apis/aws_provider_spec.go

```diff
@@ -103,6 +103,8 @@ type AWSProviderSpec struct {
 
 	// Tags to be specified on the EC2 instances
 	Tags map[string]string `json:"tags,omitempty"`
+
+	InstanceMetadataOptions *InstanceMetadataOptions `json:"instanceMetadataOptions,omitempty"`
```

Comment:
why do we need to define our own types and later copy values from here to the type provided by ec2? 
We have done this way earlier also but I don't get the reasoning. Is it to make sure that if a backward incompatible change happens to their struct then we don't have to make a lot of changes, like in extensions ?

Reply:
I think it makes sense to decouple the two specs. We may want to use custom json decoding, do some kind of translation in-between our spec and ec2 client, control the default behavior etc. This is also an "extreme" case where we want to expose all fields from the ec2 struct (which doesn't always make sense). 

In general I am in favor of having control over the API we expose even if inadvertently it will approximate the ec2 one.

Reply:
@unmarshall @rishabh-11 also had the same doubt. Do you guys agree with @kon-angelo's comment or still think that we should couple them.

Reply:
When i look at the `AWSProviderSpec` i see other instances where another struct has been defined. Example `AWSNetworkInterfaceSpec` which holds the network interface spec fields which gardener configures. This is then used to create `ec2.InstanceNetworkInterfaceSpecification`. Similar is the case elsewhere as well. So what @kon-angelo has done is not out of the ordinary.

I think it is fine to declare our own structs. These structs will then represent fields that are today managed (set/configured) by extensions.